src/js/game/LevelMVC/LevelView.js

```diff
@@ -573,7 +576,18 @@ module.exports = class LevelView {
   playOpenChestAnimation(position) {
     const blockIndex = (this.yToIndex(position[1])) + position[0];
     const block = this.actionPlaneBlocks[blockIndex];
-    this.playScaledSpeed(block.animations, "open");
+    const animation = this.playScaledSpeed(block.animations, "open");
+    this.onAnimationEnd(animation, () => {
+      let blockData = this.controller.levelModel.actionPlane.getBlockAt(position);
+      let treasure = this.controller.levelModel.actionPlane.getBlockAt(position).blockType.substring(0, blockData.blockType.length - 5);
```

Comment:
For readability it'd be nice to extract a helper to hide away the substring logic here, so it reads something like
```js
const treasure = treasureTypeFromBlock(blockData);
```